Add waiting time to gate evaluate command (#919)
DataDog/datadog-ci: src/commands/gate/README.md

- `--dry-run` runs the command without the final evaluation step. All other checks are performed. The default value is `false`.
- `--fail-if-unavailable` fails the command if Datadog is unavailable. The default value is `false`.
- `--fail-on-empty` fails the command if no matching rules are found in Datadog. The default value is `false`.

> - `--no-wait` removes the waiting time (30s) that is in place to ensure that events (e.g. tests) can be properly queried by the rules. This mechanism avoids that rules are incorrectly evaluated due to the absence of events. Read more on the quality gates documentation before passing this flag. The default value is `false`.

Reviewer: ```suggestion
- `--no-wait` removes the waiting time (30s) that is in place to ensure that events (for example: tests) can be properly queried by the rules. This mechanism prevents rules from being incorrectly evaluated due to the absence of events. Read more on the quality gates documentation before passing this flag. The default value is `false`.
```

would be nice to add a link to the quality gates documentation

Author: Thanks, I have committed your suggestion.  We still need to create the documentation, so no link for now 😬

> - `--no-wait` can be used in order to remove the default waiting time (30s) that the command has to ensure that the events (e.g. tests) can be properly queried by the rules. This mechanism avoids that rules are incorrectly evaluated due to the absence of the events. Read more on the quality gates documentation before enabling this. The default value is `false`.

Reviewer 2: I think this is a bit clearer:
```suggestion
- `--no-wait` removes the waiting time (30s) that is in place to ensure that events (e.g. tests) can be properly queried by the rules. This mechanism avoids that rules are incorrectly evaluated due to the absence of events. Read more on the quality gates documentation before passing this flag. The default value is `false`.
```

Also, why do we want to allow users to set this? Isn't it dangerous?

Author: Looks clearer, thanks! 

> Also, why do we want to allow users to set this? Isn't it dangerous?

It's not really dangerous, what can happen is that rules might be incorrectly evaluated, but we need to do a good job on the documentation side on making sure they understand the risks. There are many use cases in which they wouldn't want to wait for 30s, for example when there are intermediate jobs between when the events are sent and when `datadog-ci gate evaluate` is called.
